// remote_execution.proto
syntax = "proto3";

package remote;

message FunctionRequest {
    string function_name = 1;
    string function_code = 2;
    repeated string args = 3;
    map<string, string> kwargs = 4;
}

message FunctionResponse {
    string result = 1;
    bool success = 2;
    string error = 3;
}

service RemoteExecutor {
    rpc ExecuteFunction (FunctionRequest) returns (FunctionResponse) {}
}